src/ru/nsu/ccfit/zuev/osu/MainScene.java

```diff
@@ -1061,6 +1064,27 @@ public void run() {
         }
     }
 
+    public void showExitDialog() {
+        GlobalManager.getInstance().getMainActivity().runOnUiThread(new Runnable() {
+            public void run() {
+                AlertDialog.Builder builder = new AlertDialog.Builder(GlobalManager.getInstance().getMainActivity());
+                builder.setMessage("Are you sure you want to exit the game?");
+                builder.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        exit();
+                        PowerManager.WakeLock wakeLock = GlobalManager.getInstance().getMainActivity().getWakeLock();
+                        if (wakeLock != null && wakeLock.isHeld()) {
+                            wakeLock.release();
+                        }
+                    }
+                });
+                builder.setNegativeButton("No", null);
+                builder.show();
```

Comment:
Do not use hardcode strings, you should use resource id instead.

Reply:
okay, will do right ahead